### DISCUSSION TITLE
VoiceChat.members does not output user who is in VoiceChannel before running the script

### DISCUSSION THREAD

**Discussion by Omastto1** [Rapptz/discord.py#5875] [Q&A] [answered]
Hello, 
newbie to Discord Bot developing.
I am trying to develop a bot that records users' presence in voice chat using the task.
However, the bot does not output users which were in VoiceChannel before its script was started.

Currently, I am the only one on my test server.

Originally I thought that issue is caused by this:
_If I have been on VoiceChat before command `!start`, which starts the task, I have to reconnect to VoiceChat, in order to have my presence recorder by bot._  

However, I found out that the issue is more likely connected to frequent running and stopping the script of the bot as a whole instead of running the task itself.

The issue does not happen if I am not connected to voice chat before I run the bot script. 
As opposite, if I connect to voice chat and only after that I run the bot script. The bot does not see me and I have to reconnect to voice chat.

I understand that this is not a problem in general, because most bots work 24/7, but i would be grateful if You could say whether this is some edge case bug or explain it to me.

Thank you.

Here is part of my code:

```
def get_voice_channel_members(self):
    voice_channel = self.bot.get_channel(self.voice_channel_id)
    if voice_channel:
        return voice_channel.members
    if not voice_channel:
        return []

@tasks.loop(seconds=10)
async def my_background_task(self):
    attendees = self.get_voice_channel_members()
    for attendee in attendees:
        self.attendance[attendee.name] = self.attendance[attendee.name] + 1 if attendee.name in self.attendance else 1

@commands.command(pass_context=True)
async def start(self, ctx):
    self.attendance = dict()
    self.my_background_task.start()
```

**Comment by Rapptz** [comment, marked as answer]
In order to get help in this regard more information is needed, mainly:

1. Your discord.py version info (you can get this by doing `py -3 -m discord --version` in Windows or `python3 -m discord --version` on Linux/Mac).
2. If you're on v1.5 the intents (`discord.Intents`) that you're passing into the bot.

**Reply by Omastto1** [reply]
The issue was caused by my unfamiliarity with `discord.Intents` since I am a newbie and I did not read the documentation properly yet.
Thank You for pointing this out!
